app/src/main/java/com/jerboa/ui/components/common/TextBadge.kt

```diff
@@ -33,12 +42,61 @@ fun TextBadge(
         Text(
             text = text,
             style = textStyle,
-            overflow = TextOverflow.Clip,
             maxLines = 1,
             color = textColor,
             modifier =
                 Modifier
-                    .padding(horizontalTextPadding.dp, verticalTextPadding.dp),
+                    .padding(horizontalTextPadding.dp, verticalTextPadding.dp)
+                    .basicMarquee(),
         )
     }
 }
+
+/**
+ * Displays activitypub items (communities, users), with a smaller @instance shown
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ItemAndInstanceTitle(
+    modifier: Modifier = Modifier,
+    title: String,
+    actorId: String?,
+    local: Boolean,
+    itemColor: Color = MaterialTheme.colorScheme.primary,
+    itemStyle: TextStyle = MaterialTheme.typography.bodyMedium,
+    instanceColor: Color = MaterialTheme.colorScheme.onSurface.muted,
+    instanceStyle: TextStyle = MaterialTheme.typography.bodySmall.copy(
+        fontSize = INSTANCE_FONT_SIZE.sp,
+    ),
+) {
+    val text = remember(title + local + itemColor) {
```

Comment:
shouldn't this be commas?

Reply:
Changed, I don't think it matters tho, just a string key.